### PR TITLE
share error information, inform of -v option

### PR DESCRIPTION
**Pull request by JeffreyFredrick**

### CODE REVIEW

**Review by scarytom — COMMENTED** [review]
why mess about with set +e at all? Why not just
if [ wget ... ]; then
    fail 'error logging in...'
fi

**Review by scarytom — COMMENTED** [review]
I'm not sure about this no-check-cert stuff.  Wouldn't it be better to work out why your personal wget doesn't like the perfectly valid cert from ADP?

File: payslip.sh [hunk 1]
```diff
@@ -183,6 +183,10 @@ TOKEN="$(wget ${VERBOSE} -O - \
   'https://fress2.adp.com/core/coreControl.asp?ProductType=0' \
   | grep sessionToken | cut -d "'" -f2)"
 
+if [ ${#TOKEN} = 0 ]; then
+   fail 'sessionToken not found in cookies.txt'
```
Comment (scarytom) [inline payslip.sh]: Technically, this error message is wrong -- the session token is never in cookies it is in the response body of coreControl.asp

File: payslip.sh [hunk 2]
```diff
@@ -183,6 +183,10 @@ TOKEN="$(wget ${VERBOSE} -O - \
   'https://fress2.adp.com/core/coreControl.asp?ProductType=0' \
   | grep sessionToken | cut -d "'" -f2)"
 
+if [ ${#TOKEN} = 0 ]; then
```
Comment (scarytom) [inline payslip.sh]: why not 
`[ -z "${TOKEN}" ] && fail 'unable to get session token'`

**Comment by JeffreyFredrick** [comment]
_why mess about with set +e at all?_
Because I wanted to pass the wget error code back to the shell; the value has information.

**Comment by JeffreyFredrick** [comment]
_I'm not sure about this no-check-cert stuff._
I'm not sure either, which is why I made it separate commit, and had a comment that it probably shouldn't be used. However if you turn on verbose output you are informed by wget about the option. That being the case I thought it was worth supplying the option.

File: payslip.sh (hunk 1 above)
Comment (JeffreyFredrick) [reply payslip.sh]: Updated error message. Directing to look for actual values in cookies.txt. The missing values are a symptom I see when I test authentication errors.

File: payslip.sh (hunk 2 above)
Comment (JeffreyFredrick) [reply payslip.sh]: I've no preference between `[ ${#token} = 0 ]` and `[ -z "${TOKEN}" ]`; I'm equally unfamiliar with either. Is there a reason to prefer one over the other?

Comment (JeffreyFredrick) [reply payslip.sh]: I don't like using `&&` as a way of controlling execution. I think it is less clear than using the `if`.

Comment (scarytom) [reply payslip.sh]: I guess in both cases the reasoning is based around the normal practice.  Length checking is more obscure than performing the empty test. Similarly, for one-line check-and-exit tests it is common practice to use the && for execution control in shell. I don't find it harder to parse, but I don't have as strong feelings on this as many will.

File: payslip.sh (hunk 1 above)
Comment (scarytom) [reply payslip.sh]: I'm not sure directing to cookies.txt is fair, as that is akin to asking a user to look in a private variable of the code -- cookies.txt is internal, and deleted after program execution.  We should really be using a trap to ensure that cookies.txt is always deleted, even in the event of failure, as it is a security leak to leave it lying around.

**Comment by scarytom** [comment]
> I'm not sure either, which is why I made it separate commit,

Yeah, that isn't how pull requests work. I can only merge the all the commits or none. Really, separate changes should all be done on separate pull requests.

**Comment by scarytom** [comment]
> Because I wanted to pass the wget error code back to the shell; the value has information.

In which case, you should probably do 
 `wget ... || error = $?`

https://stackoverflow.com/questions/18621990/bash-get-exit-status-of-command-when-set-e-is-active

**Comment by JeffreyFredrick** [comment]
Thanks for the comments. I appreciate feedback!

I'll make some changes including the -z and removing the set +e. I'm still torn on the no-check-cert... but I'll probably pull it.

**Comment by JeffreyFredrick** [comment]
fwiw the whole idea of checking the error code from wget was with the idea I could give different error messages to distinguish authentication errors from the cert errors. that turned out to not be possible (thanks ADP!) which made that kind of pointless.

File: payslip.sh (hunk 1 above)
Comment (JeffreyFredrick) [reply payslip.sh]: I'll change the message to remove the reference to cookies.txt

File: payslip.sh (hunk 2 above)
Comment (JeffreyFredrick) [reply payslip.sh]: changed to empty test. not adopting the idiomatic use of `&&`.

**Comment by adamgibbins** [comment]
[I merged](https://github.com/scarytom/payslip-retriever/commit/dd7c629f4928c565c84785a5e17f09a14a0c136b) the one agreed change in this PR (which would have detected the most recent regression...). Closing this now, feel free to open a PR for any additional alterations and we can discuss there.

**Pull request closed without merging**